# Texas Real Estate Investment Analyzer
**Week 4 Capstone Project**

## Problem Statement
Real estate investors spend hours manually comparing neighborhoods, calculating ROI metrics, and searching for listings that match their criteria. This tool automates that process - taking budget, strategy (cash flow vs appreciation), and location inputs to return ranked neighborhoods with example properties in seconds.

## Tool Purpose
- Help investors make data-driven decisions faster
- Provide clear ROI comparisons (LTR vs STR for cash flow, growth metrics for appreciation)
- Surface actionable listings in top-performing neighborhoods


In [22]:
# Import libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from scipy import stats

# Display settings
pd.set_option('display.max_columns', None)
pd.set_option('display.width', None)

print("Libraries loaded successfully!")

Libraries loaded successfully!


In [23]:
# Load Zillow ZHVI data
df_home_values = pd.read_csv('../data/raw/zillow_zhvi_austin.csv')

# Explore structure
print("Dataset shape:", df_home_values.shape)
print("\\n" + "="*50)
df_home_values.head()

Dataset shape: (895, 314)
\n==================================================


RegionID  SizeRank       RegionName RegionType StateName     2000-01-31  \
0    102001         0    United States    country       NaN  123328.165417   
1    394913         1     New York, NY        msa        NY  222096.674116   
2    753899         2  Los Angeles, CA        msa        CA  222620.175787   
3    394463         3      Chicago, IL        msa        IL  155857.310945   
4    394514         4       Dallas, TX        msa        TX  128023.642757   

      2000-02-29     2000-03-31     2000-04-30     2000-05-31     2000-06-30  \
0  123545.139196  123814.218630  124391.341197  125055.540080  125767.650802   
1  223040.459669  223992.986378  225923.174072  227921.950272  230137.888325   
2  223448.605299  224552.065508  226747.580645  229148.785908  231533.175208   
3  156001.589412  156276.370294  156959.956841  157782.229234  158705.041331   
4  128080.664935  128146.217766  128316.451582  128540.900207  128768.088559   

      2000-07-31     2000-08-31     2000-09-30     2000-10-31     2000-11-30  \
0  126521.915537  127336.070915  128182.937049  129033.028662  129879.049429   
1  232558.855927  235137.318716  237776.090493  240408.028124  242959.724600   
2  233977.414207  236444.939491  238769.327713  240844.888732  242721.426393   
3  159687.250807  161073.183668  162700.595698  164575.314240  165997.475062   
4  129006.292438  129363.119882  129810.298811  130319.050194  130798.348071   

      2000-12-31     2001-01-31     2001-02-28     2001-03-31     2001-04-30  \
0  130726.212923  131503.860487  132192.553266  132856.808358  133575.015806   
1  245369.405408  247665.531326  250020.000099  252304.951735  254615.389509   
2  244599.166756  245966.951387  247455.409073  249146.610940  251633.141028   
3  167159.395369  167998.337014  168820.830692  169685.760103  170603.649599   
4  131259.048064  131493.778492  131301.265764  131018.367821  130859.075526   

      2001-05-31     2001-06-30     2001-07-31     2001-08-31     2001-09-30  \
0  134350.071383  135145.261752  135963.483531  136818.280548  137680.702974   
1  256886.093194  259331.549197  261823.763707  264500.210360  267261.631357   
2  254229.529269  256711.046966  259203.608714  261696.033827  264188.519861   
3  171708.048909  172858.862250  173988.335209  175383.070780  176946.331303   
4  130973.477662  131074.231188  131175.130665  131429.293539  131678.995344   

      2001-10-31     2001-11-30     2001-12-31     2002-01-31     2002-02-28  \
0  138539.419698  139339.787970  140090.149100  140753.774955  141388.519329   
1  270235.945046  273101.232425  275857.425080  278588.522592  281206.406822   
2  266602.895108  268752.615012  270891.499190  272584.984253  274557.080561   
3  178789.459694  180157.214578  181213.781283  181901.259237  182722.010931   
4  131896.751604  131985.992132  132090.580128  132172.983306  132363.690242   

      2002-03-31     2002-04-30     2002-05-31     2002-06-30     2002-07-31  \
0  142035.612681  142739.625135  143517.220238  144359.510310  145282.909005   
1  283695.176216  286059.906225  288523.652591  291312.482781  294397.399703   
2  277001.102793  280308.654008  284116.953677  288120.624293  292912.916701   
3  183685.869564  184607.921353  185604.588958  186611.539229  187824.074247   
4  132652.384993  133124.322202  133622.362305  134267.559256  134905.750579   

      2002-08-31     2002-09-30     2002-10-31     2002-11-30     2002-12-31  \
0  146268.793246  147267.789851  148265.276426  149215.313181  150140.201911   
1  298117.762139  302162.130628  306293.847190  310120.740653  313876.044319   
2  298009.837669  303194.022277  307938.878598  312516.390019  316992.362489   
3  189360.176856  191036.168763  192833.397800  194048.109939  195127.690853   
4  135559.080928  136085.315163  136533.815697  136929.374451  137277.750422   

      2003-01-31     2003-02-28     2003-03-31     2003-04-30     2003-05-31  \
0  150977.179563  151778.047825  152578.482461  153449.900991  154408.684905   
1  3176

In [24]:
# Check data types and missing values
df_home_values.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 895 entries, 0 to 894
Columns: 314 entries, RegionID to 2025-09-30
dtypes: float64(309), int64(2), object(3)
memory usage: 2.1+ MB


In [25]:
# Statistical summary
df_home_values.describe()

RegionID    SizeRank     2000-01-31     2000-02-29     2000-03-31  \
count     895.000000  895.000000     431.000000     432.000000     433.000000   
mean   412099.672626  461.751955  110704.972129  110884.115433  111041.702120   
std     78377.355083  268.710532   46877.162590   46989.971606   47181.617799   
min    102001.000000    0.000000   44273.377963   44298.984636   44402.262943   
25%    394546.000000  230.500000   79749.400803   79870.411382   80006.168222   
50%    394795.000000  460.000000   99093.472236   99534.169481   99340.233458   
75%    395044.500000  689.500000  127625.660548  127749.081161  127998.441120   
max    753929.000000  939.000000  361937.104143  363802.496463  366612.675871   

          2000-04-30     2000-05-31     2000-06-30     2000-07-31  \
count     435.000000     437.000000     438.000000     439.000000   
mean   111739.394122  112476.812577  113047.491682  113838.167384   
std     47850.606628   48447.380198   49070.391660   49802.061550   
min     44604.376607   44880.514808   45216.748068   45549.144544   
25%     80371.028033   80711.431085   81260.912406   81599.958909   
50%     99607.376674  100625.380188  101000.309958  101251.252636   
75%    128432.719447  129201.780987  129871.941118  130516.753400   
max    374222.843543  382201.638094  391343.415051  399213.659211   

          2000-08-31     2000-09-30     2000-10-31     2000-11-30  \
count     440.000000     441.000000     441.000000     444.000000   
mean   114465.503741  115158.009905  115854.148879  116575.052396   
std     50542.821177   51293.660106   52098.699978   52732.868096   
min     45796.795977   45975.969456   46111.049748   46269.230240   
25%     81856.839877   82377.612309   82620.690500   83048.063602   
50%    101340.035596  101639.703737  102517.290598  103169.271843   
75%    130968.119745  131396.131821  132600.582233  133218.140353   
max    408901.068890  418717.838293  428707.219741  438943.052021   

          2000-12-31     2001-01-31     2001-02-28     2001-03-31  \
count     445.000000     453.000000     453.000000     453.000000   
mean   117414.206774  118599.919499  119304.108366  119981.787053   
std     53441.127133   53960.713275   54672.365658   55442.363291   
min     46422.321206   46625.818996   46829.890451   47046.735119   
25%     83424.424861   84266.031756   84568.107229   84997.678637   
50%    104265.108498  105024.996165  105170.989490  105354.192079   
75%    134474.862031  136091.400130  136958.732093  137931.370963   
max    448427.154327  457907.506525  466697.206909  475810.472139   

          2001-04-30     2001-05-31     2001-06-30     2001-07-31  \
count     455.000000     456.000000     457.000000     458.000000   
mean   120544.183001  121243.624504  122054.640432  122802.142045   
std     56139.535109   56823.553229   57470.159718   58013.060764   
min     47278.786506   47560.298715   47885.146669   48262.534283   
25%     85376.018293   85851.911494   86234.424371   86695.394763   
50%    105760.656935  106362.795023  106961.023455  107731.797764   
75%    138527.770830  139538.421577  140822.791241  141642.246510   
max    483941.919036  489675.251007  491647.613154  491421.605648   

          2001-08-31     2001-09-30     2001-10-31     2001-11-30  \
count     458.000000     458.000000     459.000000     460.000000   
mean   123474.514059  124147.599186  124789.131757  125424.977330   
std     58602.737131   59181.437762   59632.502618   60025.890936   
min     48557.080092   48764.460558   48880.439065   49133.183659   
25%     86959.061672   87325.580514   87675.692857   87925.025818   
50%    108079.188162  108260.768224  108970.234386  109596.209258   
75%    142078.920135  143041.081890  143462.939473  143771.104840   
max    490378.713182  489024.991159  486076.090886  480912.266407   

          2001-12-31     2002-01-31     2002-02-28     2002-03-31  \
count     461.000000     462.000000     462.000000     462.000000   
mean   125897.100024  126270.19526

In [26]:
# Load Zillow ZORI data
df_rent = pd.read_csv('../data/raw/zillow_zori_austin.csv')

# Explore structure
print("Dataset shape:", df_rent.shape)
print("\\n" + "="*50)
df_rent.head()

Dataset shape: (695, 134)
\n==================================================


RegionID  SizeRank       RegionName RegionType StateName   2015-01-31  \
0    102001         0    United States    country       NaN  1187.699708   
1    394913         1     New York, NY        msa        NY  2339.775516   
2    753899         2  Los Angeles, CA        msa        CA  1791.360326   
3    394463         3      Chicago, IL        msa        IL  1371.236562   
4    394514         4       Dallas, TX        msa        TX  1085.958962   

    2015-02-28   2015-03-31   2015-04-30   2015-05-31   2015-06-30  \
0  1193.115753  1198.991436  1203.873957  1208.230005  1212.375541   
1  2355.226733  2369.036436  2379.004985  2383.056204  2387.391302   
2  1800.825788  1811.997912  1822.454810  1831.062613  1839.727225   
3  1372.817260  1377.145380  1381.157002  1385.580401  1388.793862   
4  1089.748669  1093.537441  1100.096365  1104.649473  1109.843504   

    2015-07-31   2015-08-31   2015-09-30   2015-10-31   2015-11-30  \
0  1216.925956  1222.043841  1227.371030  1232.602488  1237.539006   
1  2392.622804  2401.338930  2409.894440  2418.046348  2422.482283   
2  1848.845803  1859.744563  1872.268330  1882.983990  1895.093429   
3  1391.458339  1395.224359  1398.446549  1403.061354  1407.522960   
4  1114.306107  1118.937288  1123.915451  1128.417487  1133.885906   

    2015-12-31   2016-01-31   2016-02-29   2016-03-31   2016-04-30  \
0  1242.044749  1245.687254  1248.825897  1250.844299  1254.266127   
1  2427.485438  2431.010071  2434.717760  2435.820966  2437.658733   
2  1906.448707  1917.122046  1924.933044  1930.997933  1937.653961   
3  1413.054487  1416.849985  1419.473535  1421.457066  1422.511051   
4  1139.386233  1143.195864  1147.929399  1151.728458  1157.060651   

    2016-05-31   2016-06-30   2016-07-31   2016-08-31   2016-09-30  \
0  1257.878684  1262.578442  1266.374633  1270.362790  1274.049119   
1  2440.740747  2445.002423  2450.982743  2454.160492  2458.870980   
2  1946.624778  1956.625917  1966.797628  1974.514663  1982.849721   
3  1424.345869  1427.024219  1432.112515  1438.214942  1440.921257   
4  1160.269510  1164.278014  1168.112997  1173.330134  1178.017939   

    2016-10-31   2016-11-30   2016-12-31   2017-01-31   2017-02-28  \
0  1277.889974  1281.293473  1284.582468  1289.391953  1294.951861   
1  2463.718017  2473.557196  2477.135750  2480.130954  2479.918674   
2  1990.376345  1998.073665  2005.660460  2018.149262  2031.571340   
3  1443.748955  1443.167057  1442.882590  1444.545707  1450.449524   
4  1183.528299  1186.937680  1190.900644  1196.924989  1201.772554   

    2017-03-31   2017-04-30   2017-05-31   2017-06-30   2017-07-31  \
0  1301.831218  1306.818727  1311.716605  1315.572034  1319.664314   
1  2483.839014  2487.030048  2492.727152  2495.728727  2499.307524   
2  2047.758070  2058.609571  2068.041236  2074.936426  2082.933059   
3  1458.484860  1464.344283  1468.719366  1470.131952  1471.378880   
4  1209.367327  1213.227420  1220.973406  1224.229390  1228.636471   

    2017-08-31   2017-09-30   2017-10-31   2017-11-30   2017-12-31  \
0  1323.107967  1327.790875  1331.695924  1336.362078  1340.973372   
1  2500.279406  2503.581574  2503.028546  2503.468365  2505.918231   
2  2089.795463  2098.076313  2104.840543  2114.255685  2122.971442   
3  1470.884509  1475.196371  1477.190239  1481.548233  1484.624148   
4  1231.979532  1236.088623  1238.061389  1243.275097  1247.035924   

    2018-01-31   2018-02-28   2018-03-31   2018-04-30   2018-05-31  \
0  1346.005334  1351.056500  1355.239672  1359.717229  1363.446617   
1  2511.317753  2516.668436  2521.168290  2528.629061  2533.821297   
2  2132.866030  2143.050057  2148.198319  2154.676762  2158.822955   
3  1488.228366  1491.808513  1492.909505  1495.284675  1497.768792   
4  1252.685533  1257.088430  1258.705833  1261.632997  1262.440937   

    2018-06-30   2018-07-31   2018-08-31   2018-09-30   2018-10-31  \
0  1367.898302  1372.522532  1377.667810  1381.924837  1387.152312   
1  2538.383839  2543.036297  2551.0601

In [27]:
# Check data types and missing values
df_rent.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 695 entries, 0 to 694
Columns: 134 entries, RegionID to 2025-09-30
dtypes: float64(129), int64(2), object(3)
memory usage: 727.7+ KB


In [28]:
# Statistical summary
df_rent.describe()

RegionID    SizeRank   2015-01-31   2015-02-28   2015-03-31  \
count     695.000000  695.000000   212.000000   214.000000   217.000000   
mean   415261.618705  371.435971  1021.334234  1024.919238  1028.089176   
std     88278.591966  233.014875   300.097718   301.090955   303.690470   
min    102001.000000    0.000000   607.312753   610.551375   573.478876   
25%    394545.000000  175.500000   839.686991   842.850611   842.807671   
50%    394800.000000  352.000000   956.593378   962.961971   966.608464   
75%    395052.000000  545.000000  1097.539372  1100.267269  1103.841399   
max    845172.000000  921.000000  2556.384876  2570.556729  2590.039818   

        2015-04-30   2015-05-31   2015-06-30   2015-07-31   2015-08-31  \
count   222.000000   222.000000   224.000000   223.000000   224.000000   
mean   1027.078384  1030.422059  1031.299811  1035.201958  1039.560305   
std     306.404533   307.570268   308.616413   311.345683   313.382649   
min     577.955351   581.506129   585.286546   586.624277   582.722473   
25%     843.568639   849.323387   848.688432   851.553482   854.020891   
50%     968.579954   975.229193   977.647941   980.908151   981.312623   
75%    1102.834018  1107.212940  1109.323245  1109.708455  1114.580966   
max    2618.408702  2641.878095  2667.557695  2686.635010  2706.701002   

        2015-09-30   2015-10-31   2015-11-30   2015-12-31   2016-01-31  \
count   225.000000   226.000000   227.000000   233.000000   236.000000   
mean   1042.225801  1046.147740  1051.250699  1051.877765  1052.953951   
std     315.096327   317.271252   318.820267   318.434735   318.816144   
min     577.183156   571.715141   576.114398   586.748528   593.991074   
25%     849.839435   855.073768   860.379954   855.767095   858.345459   
50%     986.417173   987.071844   995.661034   993.257876   993.816197   
75%    1116.612299  1128.300826  1135.713324  1135.795181  1139.026283   
max    2715.944032  2735.304577  2749.810293  2764.343436  2767.236427   

        2016-02-29   2016-03-31   2016-04-30   2016-05-31   2016-06-30  \
count   236.000000   239.000000   242.000000   256.000000   258.000000   
mean   1055.938948  1063.523999  1069.194338  1058.835784  1061.921187   
std     321.277306   329.520895   334.086065   333.848533   334.292157   
min     589.576825   586.375819   634.221820   586.148296   589.072177   
25%     857.939729   858.583711   860.465853   857.724652   858.658007   
50%     996.526555  1000.854847  1000.818344   998.107782   994.049755   
75%    1139.299863  1145.732036  1151.960420  1130.554685  1139.850527   
max    2779.418246  2786.312281  2779.776149  2778.323731  2773.522793   

        2016-07-31   2016-08-31   2016-09-30   2016-10-31   2016-11-30  \
count   260.000000   262.000000   262.000000   261.000000   262.000000   
mean   1064.044968  1068.464385  1071.857221  1076.324007  1077.767238   
std     334.565898   335.640601   337.730315   337.806192   340.486004   
min     593.272980   596.589157   597.862400   602.334497   607.001137   
25%     858.624040   862.006162   863.866689   867.714801   866.127525   
50%     994.717238   992.576696   992.230045  1000.200016  1003.168746   
75%    1144.217664  1148.120416  1151.600626  1155.652009  1158.926818   
max    2774.674376  2759.481919  2757.033500  2750.039859  2758.350685   

        2016-12-31   2017-01-31   2017-02-28   2017-03-31   2017-04-30  \
count   266.000000   269.000000   269.000000   269.000000   270.000000   
mean   1076.370380  1078.921447  1085.174857  1088.510653  1093.060905   
std     341.026417   343.100319   343.274084   346.818950   347.546823   
min     602.484126   600.687526   604.889599   605.500718   607.702898   
25%     863.471861   867.708704   874.326023   874.658824   878.532269   
50%     996.316469  1002.912485  1005.863953  1010.487707  1014.062923   
75%    1166.561216  1173.655195  1181.922346  1178.975560  1188.698050   
max    2764.200740  2785.466869  2798.590952  2816.805163  2823.090132   

 

In [29]:
# Load Airbnb data
df_airbnb = pd.read_csv('../data/raw/airbnb_austin.csv.gz')

# Explore structure
print("Dataset shape:", df_airbnb.shape)
print("\\n" + "="*50)
df_airbnb.head()

Dataset shape: (15187, 79)
\n==================================================


id                        listing_url       scrape_id last_scraped  \
0  5456  https://www.airbnb.com/rooms/5456  20250613040113   2025-06-13   
1  5769  https://www.airbnb.com/rooms/5769  20250613040113   2025-06-13   
2  6413  https://www.airbnb.com/rooms/6413  20250613040113   2025-06-14   
3  6448  https://www.airbnb.com/rooms/6448  20250613040113   2025-06-13   
4  8502  https://www.airbnb.com/rooms/8502  20250613040113   2025-06-13   

            source                                               name  \
0      city scrape          Walk to 6th, Rainey St and Convention Ctr   
1      city scrape                                     NW Austin Room   
2  previous scrape                      Gem of a Studio near Downtown   
3      city scrape  Secluded Studio @ Zilker - King Bed, Bright & ...   
4      city scrape                            Woodland Studio Lodging   

                                         description  \
0  Great central  location for walking to Convent...   
1                                                NaN   
2  Great studio apartment, perfect a single perso...   
3  Clean, private space with everything you need ...   
4  Studio rental on lower level of home located i...   

                               neighborhood_overview  \
0  My neighborhood is ideally located if you want...   
1  Quiet neighborhood with lots of trees and good...   
2  Travis Heights is one of the oldest neighborho...   
3  The neighborhood is fun and funky (but quiet)!...   
4                                                NaN   

                                         picture_url  host_id  \
0  https://a0.muscache.com/pictures/14084884/b5a3...     8028   
1  https://a0.muscache.com/pictures/23822033/ac94...     8186   
2  https://a0.muscache.com/pictures/hosting/Hosti...    13879   
3  https://a0.muscache.com/pictures/airflow/Hosti...    14156   
4  https://a0.muscache.com/pictures/miso/Hosting-...    25298   

                                  host_url  host_name  host_since  \
0   https://www.airbnb.com/users/show/8028     Sylvia  2009-02-16   
1   https://www.airbnb.com/users/show/8186  Elizabeth  2009-02-19   
2  https://www.airbnb.com/users/show/13879       Todd  2009-04-17   
3  https://www.airbnb.com/users/show/14156        Amy  2009-04-20   
4  https://www.airbnb.com/users/show/25298      Karen  2009-07-11   

  host_location                                         host_about  \
0    Austin, TX  I am a licensed Real Estate Broker and owner o...   
1    Austin, TX  We're easygoing professionals that enjoy meeti...   
2    Austin, TX  We're a young family that likes to travel, we ...   
3    Austin, TX  We are a family of four (with teenagers, all o...   
4    Austin, TX  I handle the reservations at the studio on the...   

   host_response_time host_response_rate host_acceptance_rate  \
0  within a few hours               100%                  92%   
1      within an hour               100%                 100%   
2      within an hour               100%                 100%   
3      within an hour               100%                  96%   
4        within a day                80%                  50%   

  host_is_superhost                                 host_thumbnail_url  \
0                 t  https://a0.muscache.com/im/users/8028/profile_...   
1                 f  https://a0.muscache.com/im/users/8186/profile_...   
2                 t  https://a0.muscache.com/im/pictures/user/4f35e...   
3                 t  https://a0.muscache.com/im/users/14156/profile...   
4                 f  https://a0.muscache.com/im/users/25298/profile...   

                                    host_picture_url host_neighbourhood  \
0  https://a0.muscache.com/im/users/8028/profile_...      East Downtown   
1  https://a0.muscache.com/im/users/8186/profile_...  SW Williamson Co.   
2  https://a0.muscache.com/im/pictures/user/4f35e...     Travis Heights   
3  https://a0.muscache.com/im/users/14156/profile...             Zilker   
4  

In [30]:
# Check data types and missing values
df_airbnb.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15187 entries, 0 to 15186
Data columns (total 79 columns):
 #   Column                                        Non-Null Count  Dtype  
---  ------                                        --------------  -----  
 0   id                                            15187 non-null  int64  
 1   listing_url                                   15187 non-null  object 
 2   scrape_id                                     15187 non-null  int64  
 3   last_scraped                                  15187 non-null  object 
 4   source                                        15187 non-null  object 
 5   name                                          15187 non-null  object 
 6   description                                   14840 non-null  object 
 7   neighborhood_overview                         7914 non-null   object 
 8   picture_url                                   15186 non-null  object 
 9   host_id                                       15187 non-null 

In [31]:
# Statistical summary (numeric columns only)
df_airbnb.describe()

id     scrape_id       host_id  host_listings_count  \
count  1.518700e+04  1.518700e+04  1.518700e+04         15185.000000   
mean   5.995245e+17  2.025061e+13  1.817908e+08           120.095357   
std    5.520278e+17  7.531498e+00  2.008631e+08           575.928485   
min    5.456000e+03  2.025061e+13  2.300000e+01             1.000000   
25%    3.215426e+07  2.025061e+13  1.849321e+07             1.000000   
50%    6.973160e+17  2.025061e+13  8.476714e+07             2.000000   
75%    1.128970e+18  2.025061e+13  3.327081e+08            11.000000   
max    1.441661e+18  2.025061e+13  7.006501e+08          4889.000000   

       host_total_listings_count  neighbourhood_cleansed  \
count               15185.000000            15187.000000   
mean                  183.402305            78724.375979   
std                   858.930696               20.806128   
min                     1.000000            78701.000000   
25%                     1.000000            78704.000000   
50%                     3.000000            78723.000000   
75%                    15.000000            78745.000000   
max                  8202.000000            78759.000000   

       neighbourhood_group_cleansed      latitude     longitude  accommodates  \
count                           0.0  15187.000000  15187.000000  15187.000000   
mean                            NaN     30.280347    -97.750361      5.008560   
std                             NaN      0.064183      0.064474      3.426457   
min                             NaN     30.078440    -98.053350      1.000000   
25%                             NaN     30.241905    -97.768320      2.000000   
50%                             NaN     30.269239    -97.741428      4.000000   
75%                             NaN     30.309210    -97.716984      6.000000   
max                             NaN     30.518350    -97.559300     16.000000   

          bathrooms      bedrooms          beds  minimum_nights  \
count  10714.000000  14248.000000  10696.000000    15187.000000   
mean       1.738380      2.095101      2.972046        7.888984   
std        1.046643      1.441938      2.981831       30.206797   
min        0.000000      0.000000      0.000000        1.000000   
25%        1.000000      1.000000      1.000000        1.000000   
50%        1.000000      2.000000      2.000000        2.000000   
75%        2.000000      3.000000      4.000000        3.000000   
max       17.000000     23.000000    132.000000     1124.000000   

       maximum_nights  minimum_minimum_nights  maximum_minimum_nights  \
count    15187.000000            15187.000000            15187.000000   
mean       461.863172                7.339698               10.869362   
std        432.045884               30.122596               40.731182   
min          1.000000                1.000000                1.000000   
25%         60.000000                1.000000                2.000000   
50%        365.000000                2.000000                3.000000   
75%       1125.000000                3.000000                4.000000   
max       1825.000000             1124.000000             1124.000000   

       minimum_maximum_nights  maximum_maximum_nights  minimum_nights_avg_ntm  \
count            1.518700e+04            1.518700e+04             15187.00000   
mean             1.273240e+06            5.798163e+06                 8.31683   
std              5.226375e+07            1.114328e+08                30.56368   
min              1.000000e+00            1.000000e+00                 1.00000   
25%              9.900000e+01            3.600000e+02                 1.30000   
50%              3.650000e+02            3.650000e+02                 2.00000   
75%              1.125000e+03            1.125000e+03                 3.00000   
max              2.147484e+09            2.147484e+09              1124.00000   

       maximum_nights_avg_ntm  calendar_updated  availability_30  \
count            1.518700e+04        

## Initial Data Observations

### Home Values Dataset (ZHVI):
- Shape: 895 rows × 314 columns
- Time range: 2000-01-31 to 2025-09-30
- **Issue:** Includes all metros nationwide, not just Austin
- **Solution:** Filter by RegionName containing "Austin"
- Data structure: Each column is a date (time series), each row is a region

### Rental Rates Dataset (ZORI):
- Shape: 695 rows × 134 columns
- Time range: 2015-01-31 to 2025-09-30
- **Issue:** Same as ZHVI - includes all metros
- **Solution:** Filter by RegionName containing "Austin"
- Data structure: Same time-series format as ZHVI

### Airbnb Dataset:
- Shape: 15,187 rows × 79 columns
- Key columns: price, bedrooms, host_neighbourhood, property_type, availability_365, estimated_occupancy_l365d
- **Issue:** Need to verify column names and data formats
- **Good news:** Already Austin-specific (no filtering needed)
- Data structure: Each row is one Airbnb listing

### Questions for Claude:
1. After filtering Zillow data to Austin, how many neighborhoods/regions remain?
2. What's the granularity of Zillow data - neighborhoods, zip codes, or metro-level only?
3. How should we handle Airbnb price formatting if it's a string?
4. Which neighborhood column in Airbnb should we use for grouping?

In [32]:
# Filter ZHVI data to Austin metro only
# Look for "Austin" in RegionName column
df_home_values_austin = df_home_values[df_home_values['RegionName'].str.contains('Austin', case=False, na=False)]

print("Filtered to Austin metro:")
print(f"Original shape: {df_home_values.shape}")
print(f"Austin only: {df_home_values_austin.shape}")
print("\n" + "="*50)
df_home_values_austin.head()

Filtered to Austin metro:
Original shape: (895, 314)
Austin only: (2, 314)



RegionID  SizeRank  RegionName RegionType StateName     2000-01-31  \
29     394355        29  Austin, TX        msa        TX  172877.545503   
679    394354       698  Austin, MN        msa        MN   71914.187116   

        2000-02-29     2000-03-31     2000-04-30     2000-05-31  \
29   173446.017259  173966.530023  174794.113759  175419.076996   
679   72117.657894   72394.976107   73082.075267   73720.585433   

        2000-06-30     2000-07-31     2000-08-31     2000-09-30  \
29   175697.682638  175805.893359  176420.831675  177094.863020   
679   74310.948947   74647.983881   75104.356476   75534.612892   

        2000-10-31     2000-11-30     2000-12-31     2001-01-31  \
29   178154.813766  178855.448249  179446.419791  179648.674738   
679   76000.793424   76464.213520   77077.116245   77697.630938   

        2001-02-28     2001-03-31     2001-04-30     2001-05-31  \
29   179664.313772  179988.647553  180622.853515  181184.273068   
679   78343.893638   78885.522262   79549.395389   80211.695370   

        2001-06-30     2001-07-31     2001-08-31     2001-09-30  \
29   181557.543792  181742.161266  182005.980507  182068.916845   
679   80842.434141   81348.854519   81758.523284   82154.745617   

        2001-10-31     2001-11-30     2001-12-31     2002-01-31  \
29   182079.283103  182057.650213  181915.318738  181869.126685   
679   82581.713442   83068.843930   83586.762021   84028.246680   

        2002-02-28     2002-03-31     2002-04-30     2002-05-31  \
29   181531.268224  181379.884476  181172.499072  181020.328338   
679   84516.815399   85037.546791   85722.093610   86254.891415   

        2002-06-30     2002-07-31     2002-08-31     2002-09-30  \
29   180779.463546  180441.144496  180254.602982  180320.697062   
679   86710.699507   87005.808609   87378.119204   87740.653254   

        2002-10-31     2002-11-30     2002-12-31     2003-01-31  \
29   180339.092050  180279.490798  179894.178158  179577.995586   
679   88170.689414   88487.601995   88814.403975   89144.235980   

        2003-02-28     2003-03-31     2003-04-30     2003-05-31  \
29   179517.048645  179808.864845  180155.457302  180351.621272   
679   89552.562556   89911.192562   90272.818787   90659.143567   

        2003-06-30     2003-07-31     2003-08-31     2003-09-30  \
29   180366.880813  180585.062252  180816.947302  180888.262316   
679   91122.664299   91571.280097   92196.783866   93005.677281   

        2003-10-31     2003-11-30     2003-12-31     2004-01-31  \
29   180770.381547  180545.224886  180463.749998  180405.249280   
679   94004.759814   94689.495226   95208.968083   95512.432968   

        2004-02-29     2004-03-31     2004-04-30     2004-05-31  \
29   180916.767714  181693.655999  182576.384695  183093.591463   
679   96037.316583   96623.016778   97270.568029   97988.353573   

        2004-06-30     2004-07-31     2004-08-31     2004-09-30  \
29   183335.387115  183669.794808  184268.855457  185313.632065   
679   98720.408778   99560.486842  100173.118622  100658.644343   

        2004-10-31     2004-11-30     2004-12-31     2005-01-31  \
29   186260.235391  186914.709657  187320.582922  187616.546360   
679  100934.678293  101363.619725  101854.091394  102455.920197   

        2005-02-28     2005-03-31     2005-04-30     2005-05-31  \
29   187834.225756  187985.699241  188411.765489  189050.608040   
679  103065.334042  103670.239955  104272.649162  104715.939867   

        2005-06-30     2005-07-31     2005-08-31     2005-09-30  \
29   189949.048057  190816.885257  191543.407606  192161.271270   
679  105021.763721  105168.170345  105360.151145  105657.392715   

        2005-10-31     2005-11-30     2005-12-31     2006-01-31  \
29   192492.053312  192940.266241  193421.123077  193933.364402   
679  106033.787295  106324.391218  106572.040577  106735.655358   

        2006-02-28     2006-03-31     2006-04-30     2006-05-31  \
29   194244.650588  194258.407664  194344.596369  194545.494507   
679  1067

In [33]:
# Filter ZORI data to Austin metro only
df_rent_austin = df_rent[df_rent['RegionName'].str.contains('Austin', case=False, na=False)]

print("Filtered to Austin metro:")
print(f"Original shape: {df_rent.shape}")
print(f"Austin only: {df_rent_austin.shape}")
print("\n" + "="*50)
df_rent_austin.head()

Filtered to Austin metro:
Original shape: (695, 134)
Austin only: (2, 134)



RegionID  SizeRank  RegionName RegionType StateName   2015-01-31  \
29     394355        29  Austin, TX        msa        TX  1166.697664   
622    394354       698  Austin, MN        msa        MN          NaN   

      2015-02-28   2015-03-31   2015-04-30   2015-05-31   2015-06-30  \
29   1174.338507  1176.142024  1176.753047  1179.730347  1185.216652   
622          NaN          NaN          NaN          NaN          NaN   

      2015-07-31   2015-08-31   2015-09-30  2015-10-31   2015-11-30  \
29   1191.692343  1199.412353  1205.015639  1211.30402  1216.100367   
622          NaN          NaN          NaN         NaN          NaN   

      2015-12-31   2016-01-31  2016-02-29   2016-03-31   2016-04-30  \
29   1221.960218  1224.401922  1223.48595  1221.669888  1224.657723   
622          NaN          NaN         NaN          NaN          NaN   

      2016-05-31   2016-06-30   2016-07-31  2016-08-31   2016-09-30  \
29   1227.810097  1232.350675  1235.503166  1238.25015  1240.360543   
622          NaN          NaN          NaN         NaN          NaN   

      2016-10-31   2016-11-30   2016-12-31   2017-01-31  2017-02-28  \
29   1241.111093  1241.925273  1244.684315  1249.345756  1255.33244   
622          NaN          NaN          NaN          NaN         NaN   

      2017-03-31   2017-04-30   2017-05-31   2017-06-30   2017-07-31  \
29   1262.870623  1266.453917  1268.253516  1268.761204  1267.737935   
622          NaN          NaN          NaN          NaN          NaN   

      2017-08-31   2017-09-30   2017-10-31   2017-11-30   2017-12-31  \
29   1266.212241  1265.647709  1267.146672  1272.305978  1275.018912   
622          NaN          NaN          NaN          NaN          NaN   

     2018-01-31   2018-02-28  2018-03-31   2018-04-30   2018-05-31  \
29   1276.38777  1281.121992  1282.25931  1287.355417  1290.339846   
622         NaN          NaN         NaN          NaN          NaN   

      2018-06-30   2018-07-31   2018-08-31   2018-09-30   2018-10-31  \
29   1297.612479  1303.599966  1309.498064  1314.837301  1320.941572   
622          NaN          NaN          NaN          NaN          NaN   

      2018-11-30   2018-12-31   2019-01-31  2019-02-28   2019-03-31  \
29   1328.858459  1332.934882  1336.980748  1340.45365  1346.094556   
622          NaN          NaN          NaN         NaN          NaN   

      2019-04-30   2019-05-31   2019-06-30   2019-07-31   2019-08-31  \
29   1349.779841  1356.076265  1359.380855  1364.837755  1370.759433   
622          NaN          NaN          NaN          NaN          NaN   

     2019-09-30   2019-10-31   2019-11-30   2019-12-31  2020-01-31  \
29    1376.9426  1381.818836  1383.553727  1389.213714   1395.4162   
622         NaN          NaN          NaN          NaN         NaN   

      2020-02-29  2020-03-31   2020-04-30   2020-05-31  2020-06-30  \
29   1400.307953   1405.0708  1398.102402  1387.131727  1375.08334   
622          NaN         NaN          NaN          NaN         NaN   

      2020-07-31   2020-08-31   2020-09-30   2020-10-31   2020-11-30  \
29   1369.083031  1366.441846  1364.054655  1367.529425  1376.327571   
622          NaN          NaN          NaN          NaN          NaN   

      2020-12-31   2021-01-31   2021-02-28   2021-03-31  2021-04-30  \
29   1386.718682  1400.618091  1413.764046  1429.350221  1451.68893   
622          NaN          NaN          NaN          NaN         NaN   

      2021-05-31   2021-06-30   2021-07-31   2021-08-31   2021-09-30  \
29   1482.768015  1519.889855  1558.062633  1599.110187  1635.367402   
622          NaN          NaN          NaN          NaN          NaN   

      2021-10-31   2021-11-30   2021-12-31   2022-01-31   2022-02-28  \
29   1660.168482  1674.099389  1688.278111  1698.051019  1712.711538   
622          NaN          NaN          NaN          NaN          NaN   

      2022-03-31   2022-04-30   2022-05-31   2022-06-30   2022-07-31  \
29   1722.633642  1739.181883  1750.577592  1763.501878  177

In [34]:
# Inspect key Airbnb columns
airbnb_key_cols = ['id', 'name', 'host_neighbourhood', 'neighbourhood_cleansed', 
                   'property_type', 'price', 'bedrooms', 'beds', 'availability_365', 
                   'number_of_reviews', 'reviews_per_month']

# Check if these columns exist
existing_cols = [col for col in airbnb_key_cols if col in df_airbnb.columns]
print("Key columns available:")
print(existing_cols)
print("\n" + "="*50)

# Display sample
df_airbnb[existing_cols].head(10)

Key columns available:
['id', 'name', 'host_neighbourhood', 'neighbourhood_cleansed', 'property_type', 'price', 'bedrooms', 'beds', 'availability_365', 'number_of_reviews', 'reviews_per_month']



id                                               name  \
0   5456          Walk to 6th, Rainey St and Convention Ctr   
1   5769                                     NW Austin Room   
2   6413                      Gem of a Studio near Downtown   
3   6448  Secluded Studio @ Zilker - King Bed, Bright & ...   
4   8502                            Woodland Studio Lodging   
5  13035      Historic house in highly walkable East Austin   
6  22828                 Garage Apartment central SE Austin   
7  27632           Clarksville Condo - Weekly Rentals Only!   
8  40285                       OUTDOOR LIVING IN AUSTIN, TX   
9  47572  Cozy Private Room in North Austin for Longer S...   

  host_neighbourhood  neighbourhood_cleansed         property_type    price  \
0      East Downtown                   78702     Entire guesthouse  $101.00   
1  SW Williamson Co.                   78729  Private room in home   $45.00   
2     Travis Heights                   78704     Entire guesthouse      NaN   
3             Zilker                   78704     Entire guesthouse  $155.00   
4     East Riverside                   78741    Entire guest suite   $43.00   
5      East Downtown                   78702           Entire home  $145.00   
6     East Riverside                   78741     Entire guesthouse   $58.00   
7    Old West Austin                   78703          Entire condo   $95.00   
8    Northwest Hills                   78731           Entire home  $514.00   
9                NaN                   78758  Private room in home   $72.00   

   bedrooms  beds  availability_365  number_of_reviews  reviews_per_month  
0       1.0   2.0               336                711               3.59  
1       1.0   1.0                 4                304               1.65  
2       NaN   NaN                 0                123               0.65  
3       1.0   2.0               312                338               2.02  
4       1.0   1.0                98                 54               0.29  
5       2.0   2.0               301                 18               0.10  
6       1.0   1.0               301                 55               0.30  
7       1.0   1.0               294                 99               0.55  
8       2.0   2.0               315                 42               0.24  
9       1.0   1.0               224                  9               0.12

In [35]:
# Check if estimated_occupancy column exists
if 'estimated_occupancy_l365d' in df_airbnb.columns:
    print("✓ Estimated occupancy column found!")
    print("\nSample occupancy values:")
    print(df_airbnb['estimated_occupancy_l365d'].describe())
else:
    print("✗ Need to estimate occupancy from reviews_per_month")

✓ Estimated occupancy column found!

Sample occupancy values:
count    15187.000000
mean        62.136037
std         82.424938
min          0.000000
25%          0.000000
50%         18.000000
75%        108.000000
max        255.000000
Name: estimated_occupancy_l365d, dtype: float64


In [36]:
df_zillow_neighborhoods = pd.read_csv('../data/raw/zillow_zhvi_neighborhoods_national.csv')

# Filter to Austin metro neighborhoods
df_austin_neighborhoods = df_zillow_neighborhoods[df_zillow_neighborhoods['Metro'].str.contains('Austin', case=False, na=False)].copy()

print(f"Austin neighborhoods found: {len(df_austin_neighborhoods)}")
df_austin_neighborhoods[['RegionName', 'City', 'State']].head(20)

Austin neighborhoods found: 185


RegionName        City State
479      North Austin      Austin    TX
730     West Oak Hill      Austin    TX
793        Jollyville      Austin    TX
941      Windsor Park      Austin    TX
954     East Oak Hill      Austin    TX
1028  West University      Austin    TX
1097  Pleasant Valley      Austin    TX
1144    Franklin Park      Austin    TX
1368         Downtown      Austin    TX
1446        Riverside      Austin    TX
1478     Wells Branch      Austin    TX
1619    Garrison Park      Austin    TX
1659        Allandale      Austin    TX
1795         Sun City  Georgetown    TX
1801       Montopolis      Austin    TX
1806     North Burnet      Austin    TX
1854        Brentwood      Austin    TX
1920        St. Johns      Austin    TX
2013           Zilker      Austin    TX
2102     Barton Hills      Austin    TX

In [37]:
# Check what columns we have
print("Columns available:")
print(df_austin_neighborhoods.columns.tolist())

# Check time range
date_columns = [col for col in df_austin_neighborhoods.columns if col.startswith('20')]
print(f"\nTime range: {date_columns[0]} to {date_columns[-1]}")
print(f"Total months: {len(date_columns)}")

# Check for missing values in recent data
print("\nMissing values in Sept 2025:")
print(df_austin_neighborhoods['2025-09-30'].isna().sum())

Columns available:
['RegionID', 'SizeRank', 'RegionName', 'RegionType', 'StateName', 'State', 'City', 'Metro', 'CountyName', '2000-01-31', '2000-02-29', '2000-03-31', '2000-04-30', '2000-05-31', '2000-06-30', '2000-07-31', '2000-08-31', '2000-09-30', '2000-10-31', '2000-11-30', '2000-12-31', '2001-01-31', '2001-02-28', '2001-03-31', '2001-04-30', '2001-05-31', '2001-06-30', '2001-07-31', '2001-08-31', '2001-09-30', '2001-10-31', '2001-11-30', '2001-12-31', '2002-01-31', '2002-02-28', '2002-03-31', '2002-04-30', '2002-05-31', '2002-06-30', '2002-07-31', '2002-08-31', '2002-09-30', '2002-10-31', '2002-11-30', '2002-12-31', '2003-01-31', '2003-02-28', '2003-03-31', '2003-04-30', '2003-05-31', '2003-06-30', '2003-07-31', '2003-08-31', '2003-09-30', '2003-10-31', '2003-11-30', '2003-12-31', '2004-01-31', '2004-02-29', '2004-03-31', '2004-04-30', '2004-05-31', '2004-06-30', '2004-07-31', '2004-08-31', '2004-09-30', '2004-10-31', '2004-11-30', '2004-12-31', '2005-01-31', '2005-02-28', '2005-0

In [38]:
# Get current prices (Sept 2025) for each neighborhood
df_austin_neighborhoods['current_price_2025'] = df_austin_neighborhoods['2025-09-30']

# Get peak price (find max across 2022 columns)
peak_cols = [col for col in df_austin_neighborhoods.columns if col.startswith('2022')]
df_austin_neighborhoods['peak_price_2022'] = df_austin_neighborhoods[peak_cols].max(axis=1)

# Get pre-pandemic baseline (average of 2015-2019)
baseline_cols = [col for col in df_austin_neighborhoods.columns 
                 if '2015' in col or '2016' in col or '2017' in col or '2018' in col or '2019' in col]
df_austin_neighborhoods['baseline_price_2015_2019'] = df_austin_neighborhoods[baseline_cols].mean(axis=1)

# Display summary
print("Price Summary:")
print(df_austin_neighborhoods[['RegionName', 'baseline_price_2015_2019', 'peak_price_2022', 'current_price_2025']].head(10))

Price Summary:
           RegionName  baseline_price_2015_2019  peak_price_2022  \
479      North Austin             229105.449090    473143.572750   
730     West Oak Hill             384056.110091    750091.130881   
793        Jollyville             279940.139203    561026.170890   
941      Windsor Park             353271.325325    644481.064347   
954     East Oak Hill             434945.146316    837738.558850   
1028  West University             257155.040361    372196.172110   
1097  Pleasant Valley             199141.756547    385708.994381   
1144    Franklin Park             182423.507543    403009.359470   
1368         Downtown             564958.847918    864508.527350   
1446        Riverside             304347.844693    600817.238049   

      current_price_2025  
479        355157.917570  
730        591620.958501  
793        437978.278122  
941        450649.412230  
954        670860.099100  
1028       314777.289298  
1097       266291.354847  
1144       268219.76

In [39]:
# Calculate pre-pandemic CAGR (2015-2019)
df_austin_neighborhoods['baseline_cagr'] = (
    (df_austin_neighborhoods['2019-12-31'] / df_austin_neighborhoods['2015-01-31']) ** (1/5) - 1
) * 100

# Calculate current recovery trajectory (2023-2025)
df_austin_neighborhoods['recovery_cagr'] = (
    (df_austin_neighborhoods['2025-09-30'] / df_austin_neighborhoods['2023-01-31']) ** (1/2.75) - 1
) * 100

# Distance from peak
df_austin_neighborhoods['distance_from_peak'] = (
    (df_austin_neighborhoods['current_price_2025'] / df_austin_neighborhoods['peak_price_2022']) - 1
) * 100

# Display
print("Appreciation Metrics:")
print(df_austin_neighborhoods[['RegionName', 'baseline_cagr', 'recovery_cagr', 'distance_from_peak']].head(10))

Appreciation Metrics:
           RegionName  baseline_cagr  recovery_cagr  distance_from_peak
479      North Austin      11.343479      -6.630867          -24.936544
730     West Oak Hill       5.028664      -4.405938          -21.126789
793        Jollyville       6.123306      -4.609449          -21.932648
941      Windsor Park       6.195825      -8.468050          -30.075616
954     East Oak Hill       5.534263      -3.780243          -19.920112
1028  West University       5.105116      -3.234956          -15.427048
1097  Pleasant Valley       7.693813      -9.523908          -30.960553
1144    Franklin Park      10.468946      -9.956678          -33.445774
1368         Downtown       2.212645      -7.413039          -24.441047
1446        Riverside       9.645236      -7.306314          -25.771504


In [40]:
# Clean price column (remove $ and convert to float)
df_airbnb['price_clean'] = df_airbnb['price'].replace('[\$,]', '', regex=True).astype(float)

# Calculate monthly STR income
df_airbnb['monthly_str_income'] = df_airbnb['price_clean'] * 30 * (df_airbnb['estimated_occupancy_l365d'] / 100)

# Group by neighborhood
airbnb_by_neighborhood = df_airbnb.groupby('host_neighbourhood').agg({
    'price_clean': 'median',
    'monthly_str_income': 'median',
    'bedrooms': 'median',
    'id': 'count'  # number of listings
}).reset_index()

airbnb_by_neighborhood.columns = ['neighborhood', 'median_nightly_rate', 'median_monthly_str_income', 'median_bedrooms', 'listing_count']

print(f"Airbnb neighborhoods: {len(airbnb_by_neighborhood)}")
print(airbnb_by_neighborhood.head(10))

Airbnb neighborhoods: 733
               neighborhood  median_nightly_rate  median_monthly_str_income  \
0       Academy Acres North                416.0                    15724.8   
1                Acres Home                348.5                      513.0   
2                 Allandale                179.0                     3186.0   
3                 Allendale                 96.0                     1965.6   
4             Alphabet City                  NaN                        NaN   
5                 Alta Mira                118.5                       74.7   
6  American University Park                  NaN                        NaN   
7                   Anaheim                  NaN                        NaN   
8            Anaheim Resort                350.0                     1431.0   
9                 Anastasia                174.0                      313.2   

   median_bedrooms  listing_count  
0              3.0              1  
1              3.5              

In [41]:
# Get unique neighborhood names from each dataset
zillow_neighborhoods = set(df_austin_neighborhoods['RegionName'].str.lower().str.strip())
airbnb_neighborhoods = set(airbnb_by_neighborhood['neighborhood'].dropna().str.lower().str.strip())

# Check overlap
overlap = zillow_neighborhoods & airbnb_neighborhoods
print(f"Zillow neighborhoods: {len(zillow_neighborhoods)}")
print(f"Airbnb neighborhoods: {len(airbnb_neighborhoods)}")
print(f"Overlap: {len(overlap)}")

# Show samples
print("\nZillow samples:", list(zillow_neighborhoods)[:10])
print("\nAirbnb samples:", list(airbnb_neighborhoods)[:10])
print("\nOverlapping samples:", list(overlap)[:10])

Zillow neighborhoods: 184
Airbnb neighborhoods: 732
Overlap: 75

Zillow samples: ['meadows of blackhawk', 'mountain creek', 'wooten', 'mesa ridge', 'dove creek', 'north park', 'gatlinburg', 'central east austin', 'old enfield', 'oak bluff']

Airbnb samples: ['hollows at northshore', 'crystal beach', 'mlk-183', 'wonder valley', 'manana west', 'parkridge gardens', 'the oak at twin creeks', 'village at walnut creek', 'woods of westlake heights', 'anderson mill village']

Overlapping samples: ['wells point', 'sonoma', 'eagle ridge', 'brushy bend park', 'mountain creek', 'wooten', 'bryker woods', 'chestnut', 'johnston terrace', 'heritage']


In [42]:
# Standardize neighborhood names for merging (lowercase, strip whitespace)
df_austin_neighborhoods['neighborhood_clean'] = df_austin_neighborhoods['RegionName'].str.lower().str.strip()
airbnb_by_neighborhood['neighborhood_clean'] = airbnb_by_neighborhood['neighborhood'].str.lower().str.strip()

# Merge on cleaned neighborhood names
df_merged = df_austin_neighborhoods.merge(
    airbnb_by_neighborhood,
    on='neighborhood_clean',
    how='inner'  # Only keep neighborhoods that exist in BOTH datasets
)

print(f"Merged neighborhoods: {len(df_merged)}")
print("\nMerged data preview:")
print(df_merged[['RegionName', 'current_price_2025', 'median_monthly_str_income', 'baseline_cagr', 'listing_count']].head(10))

Merged neighborhoods: 75

Merged data preview:
        RegionName  current_price_2025  median_monthly_str_income  \
0     North Austin       355157.917570                     7560.0   
1    West Oak Hill       591620.958501                     2700.0   
2     Windsor Park       450649.412230                     2019.6   
3  Pleasant Valley       266291.354847                      244.8   
4    Franklin Park       268219.760607                     2030.4   
5         Downtown       653213.588039                     5918.4   
6        Riverside       445977.598660                      522.0   
7        Allandale       769001.814422                     3186.0   
8       Montopolis       325038.409478                     2142.0   
9     North Burnet       224167.839549                     2160.0   

   baseline_cagr  listing_count  
0      11.343479              2  
1       5.028664             61  
2       6.195825            173  
3       7.693813             39  
4      10.468946       

In [43]:
# Estimate LTR monthly rent (industry rule: STR income / 2.5)
df_merged['estimated_ltr_monthly_rent'] = df_merged['median_monthly_str_income'] / 2.5

# Calculate rent-to-price ratios
df_merged['str_rent_to_price_ratio'] = (df_merged['median_monthly_str_income'] * 12 / df_merged['current_price_2025']) * 100
df_merged['ltr_rent_to_price_ratio'] = (df_merged['estimated_ltr_monthly_rent'] * 12 / df_merged['current_price_2025']) * 100

# Calculate monthly cash flow (simplified - assumes 20% down, 7% interest, 1.2% tax, 0.5% insurance, 1% maintenance)
# Monthly payment = (80% of price) * 0.007 (monthly rate) / (1 - (1 + 0.007)^-360) + taxes + insurance + maintenance
loan_amount = df_merged['current_price_2025'] * 0.8
monthly_interest_rate = 0.07 / 12
n_payments = 360
monthly_mortgage = loan_amount * (monthly_interest_rate * (1 + monthly_interest_rate)**n_payments) / ((1 + monthly_interest_rate)**n_payments - 1)
monthly_taxes = df_merged['current_price_2025'] * 0.012 / 12
monthly_insurance = df_merged['current_price_2025'] * 0.005 / 12
monthly_maintenance = df_merged['current_price_2025'] * 0.01 / 12

df_merged['monthly_costs'] = monthly_mortgage + monthly_taxes + monthly_insurance + monthly_maintenance
df_merged['str_monthly_cash_flow'] = df_merged['median_monthly_str_income'] - df_merged['monthly_costs']
df_merged['ltr_monthly_cash_flow'] = df_merged['estimated_ltr_monthly_rent'] - df_merged['monthly_costs']

print("Cash Flow Analysis:")
print(df_merged[['RegionName', 'current_price_2025', 'str_monthly_cash_flow', 'ltr_monthly_cash_flow']].head(10))

Cash Flow Analysis:
        RegionName  current_price_2025  str_monthly_cash_flow  \
0     North Austin       355157.917570            4870.595096   
1    West Oak Hill       591620.958501           -1780.002356   
2     Windsor Park       450649.412230           -1392.906605   
3  Pleasant Valley       266291.354847           -1771.669971   
4    Franklin Park       268219.760607              -0.672669   
5         Downtown       653213.588039             971.992387   
6        Riverside       445977.598660           -2855.129670   
7        Allandale       769001.814422           -2637.204690   
8       Montopolis       325038.409478            -319.327340   
9     North Burnet       224167.839549             462.506977   

   ltr_monthly_cash_flow  
0             334.595096  
1           -3400.002356  
2           -2604.666605  
3           -1918.549971  
4           -1218.912669  
5           -2579.047613  
6           -3168.329670  
7           -4548.804690  
8           -1604.527

In [44]:
# Check distribution of listing counts
print("Listing count distribution:")
print(df_merged['listing_count'].describe())
print(f"\nNeighborhoods with <10 listings: {len(df_merged[df_merged['listing_count'] < 10])}")
print(f"Neighborhoods with 10+ listings: {len(df_merged[df_merged['listing_count'] >= 10])}")

# Show neighborhoods with few listings (less reliable STR estimates)
print("\nNeighborhoods with <10 Airbnb listings (less reliable):")
print(df_merged[df_merged['listing_count'] < 10][['RegionName', 'listing_count', 'median_monthly_str_income']].sort_values('listing_count'))

Listing count distribution:
count     75.000000
mean      74.120000
std       86.431188
min        1.000000
25%        9.500000
50%       39.000000
75%      113.000000
max      407.000000
Name: listing_count, dtype: float64

Neighborhoods with <10 listings: 19
Neighborhoods with 10+ listings: 56

Neighborhoods with <10 Airbnb listings (less reliable):
                  RegionName  listing_count  median_monthly_str_income
68                 Rio Vista              1                        NaN
65            Mountain Creek              1                    15820.2
46                    Sonoma              1                    13500.0
59              Windsor Road              1                      960.0
0               North Austin              2                     7560.0
6                  Riverside              2                      522.0
66            Northtown Park              2                     1288.8
64                  Heritage              2                     3367.8
60     

In [45]:
# Keep only neighborhoods with 10+ Airbnb listings for reliable STR estimates
df_final = df_merged[df_merged['listing_count'] >= 10].copy()

print(f"Final dataset: {len(df_final)} neighborhoods with reliable data")
print("\nFinal data summary:")
print(df_final[['RegionName', 'current_price_2025', 'str_monthly_cash_flow', 'baseline_cagr', 'listing_count']].describe())

Final dataset: 56 neighborhoods with reliable data

Final data summary:
       current_price_2025  str_monthly_cash_flow  baseline_cagr  listing_count
count        5.600000e+01              56.000000      56.000000      56.000000
mean         5.693256e+05           -1999.525108       6.626428      98.000000
std          3.064154e+05            2831.478527       2.360618      88.071046
min          2.241678e+05          -15841.967681       1.590534      10.000000
25%          3.724402e+05           -2739.929460       4.979282      29.000000
50%          5.416890e+05           -1744.872150       6.226357      72.500000
75%          6.422754e+05            -615.187510       7.725070     139.250000
max          2.180725e+06            5070.504484      11.830551     407.000000


In [46]:
# CASH FLOW STRATEGY: Rank by STR monthly cash flow (descending)
df_cash_flow_ranked = df_final.sort_values('str_monthly_cash_flow', ascending=False).copy()

print("="*70)
print("TOP 10 NEIGHBORHOODS - CASH FLOW STRATEGY (STR)")
print("="*70)
print(df_cash_flow_ranked[['RegionName', 'current_price_2025', 'str_monthly_cash_flow', 
                            'ltr_monthly_cash_flow', 'str_rent_to_price_ratio', 'listing_count']].head(10))

print("\n" + "="*70)
print("TOP 10 NEIGHBORHOODS - APPRECIATION STRATEGY")
print("="*70)

# APPRECIATION STRATEGY: Rank by baseline CAGR (descending)
df_appreciation_ranked = df_final.sort_values('baseline_cagr', ascending=False).copy()
print(df_appreciation_ranked[['RegionName', 'current_price_2025', 'baseline_cagr', 
                              'recovery_cagr', 'distance_from_peak']].head(10))

TOP 10 NEIGHBORHOODS - CASH FLOW STRATEGY (STR)
           RegionName  current_price_2025  str_monthly_cash_flow  \
50     Coronado Hills       308315.641181            5070.504484   
5            Downtown       653213.588039             971.992387   
20        North Lamar       292251.581727             855.798289   
15     Georgian Acres       291676.847929             712.700414   
33  Upper Boggy Creek       575312.628599             633.091167   
9        North Burnet       224167.839549             462.506977   
54              Holly       573923.421813             442.010824   
14        Parker Lane       345078.742164             194.918845   
4       Franklin Park       268219.760607              -0.672669   
18             Wooten       420082.964771            -135.444628   

    ltr_monthly_cash_flow  str_rent_to_price_ratio  listing_count  
50             627.384484                28.821892             26  
5            -2579.047613                10.872523            181  

In [47]:
# ============================================================================
# USER INPUT SECTION
# ============================================================================

# For MVP, we'll hardcode example inputs (later: Streamlit form)

# Example Investor Profile
user_budget_min = 250000
user_budget_max = 400000
user_strategy = "Cash Flow"  # Options: "Cash Flow" or "Appreciation"
user_rental_type = "STR"  # Options: "STR" or "LTR" (only matters for Cash Flow strategy)

print("="*70)
print("TEXAS REAL ESTATE INVESTMENT ANALYZER")
print("="*70)
print(f"\nInvestor Profile:")
print(f"  Budget: ${user_budget_min:,} - ${user_budget_max:,}")
print(f"  Strategy: {user_strategy}")
if user_strategy == "Cash Flow":
    print(f"  Rental Type: {user_rental_type}")

TEXAS REAL ESTATE INVESTMENT ANALYZER

Investor Profile:
  Budget: $250,000 - $400,000
  Strategy: Cash Flow
  Rental Type: STR


In [48]:
# Filter by budget
df_budget_filtered = df_final[
    (df_final['current_price_2025'] >= user_budget_min) & 
    (df_final['current_price_2025'] <= user_budget_max)
].copy()

print(f"\nNeighborhoods matching budget: {len(df_budget_filtered)}")

if len(df_budget_filtered) == 0:
    print("\n⚠️ No neighborhoods found in this price range. Try adjusting your budget.")
else:
    # Rank based on strategy
    if user_strategy == "Cash Flow":
        if user_rental_type == "STR":
            df_recommended = df_budget_filtered.sort_values('str_monthly_cash_flow', ascending=False).head(3)
            metric_col = 'str_monthly_cash_flow'
            metric_name = "STR Monthly Cash Flow"
        else:  # LTR
            df_recommended = df_budget_filtered.sort_values('ltr_monthly_cash_flow', ascending=False).head(3)
            metric_col = 'ltr_monthly_cash_flow'
            metric_name = "LTR Monthly Cash Flow"
    else:  # Appreciation
        df_recommended = df_budget_filtered.sort_values('baseline_cagr', ascending=False).head(3)
        metric_col = 'baseline_cagr'
        metric_name = "Baseline Annual Growth"
    
    # Display recommendations
    print("\n" + "="*70)
    print(f"TOP 3 RECOMMENDATIONS - {user_strategy.upper()} STRATEGY")
    print("="*70)
    
    for i, (idx, row) in enumerate(df_recommended.iterrows(), 1):
        print(f"\n#{i}: {row['RegionName']}")
        print(f"  Price: ${row['current_price_2025']:,.0f}")
        
        if user_strategy == "Cash Flow":
            print(f"  {metric_name}: ${row[metric_col]:,.0f}/month")
            print(f"  STR Income: ${row['median_monthly_str_income']:,.0f}/month")
            print(f"  LTR Income: ${row['estimated_ltr_monthly_rent']:,.0f}/month")
            print(f"  Airbnb Listings: {row['listing_count']:.0f}")
        else:  # Appreciation
            print(f"  {metric_name}: {row[metric_col]:.2f}%")
            print(f"  Current Recovery Rate: {row['recovery_cagr']:.2f}%")
            print(f"  Distance from Peak: {row['distance_from_peak']:.1f}%")
        
        print(f"  Historical Growth: {row['baseline_cagr']:.2f}% annually (2015-2019)")


Neighborhoods matching budget: 15

TOP 3 RECOMMENDATIONS - CASH FLOW STRATEGY

#1: Coronado Hills
  Price: $308,316
  STR Monthly Cash Flow: $5,071/month
  STR Income: $7,405/month
  LTR Income: $2,962/month
  Airbnb Listings: 26
  Historical Growth: 6.70% annually (2015-2019)

#2: North Lamar
  Price: $292,252
  STR Monthly Cash Flow: $856/month
  STR Income: $3,069/month
  LTR Income: $1,228/month
  Airbnb Listings: 10
  Historical Growth: 11.42% annually (2015-2019)

#3: Georgian Acres
  Price: $291,677
  STR Monthly Cash Flow: $713/month
  STR Income: $2,921/month
  LTR Income: $1,169/month
  Airbnb Listings: 24
  Historical Growth: 11.83% annually (2015-2019)
